src/Microsoft.VisualStudio.Services.Agent/ProcessWrapper.cs

```diff
@@ -0,0 +1,295 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    /**/
+    public class ProcessWrapper
+    {
+        public event EventHandler<ProcessDataReceivedEventArgs> OutputDataReceived;
+
+        public event EventHandler<ProcessDataReceivedEventArgs> ErrorDataReceived;
+
+        public event EventHandler Exited;
+
+        private static Encoding OemEncoding = null;
+
+        public TaskCompletionSource<bool> ProcessExitedEvent = new TaskCompletionSource<bool>();
+
+        public IDictionary<String, String> EnvironmentVariables { get; private set; }
+
+        public ProcessWrapper(String workingFolder, String filename, String arguments, IDictionary<String, String> environmentVariables = null)
+        {
+            EnvironmentVariables = environmentVariables;
+            StartInfo = CreateStartInfo(workingFolder, filename, arguments, EnvironmentVariables);
+            Lock = new Object();
+        }
+
+        private static ProcessStartInfo CreateStartInfo(String workingFolder, String filename, String arguments, IDictionary<String, String> environmentVariables)
+        {
+            //     TraceLogger.Default.Info("Entering ProcessWrapper.CreateStartInfo");
+            //     TraceLogger.Default.Info("   workingFolder = {0}", workingFolder);
+            //     TraceLogger.Default.Info("   filename = {0}", filename);
+            //     TraceLogger.Default.Info("   arguments = {0}", arguments);
+
+            String workingDirectory = workingFolder;
+
+            ProcessStartInfo processStartInfo = new ProcessStartInfo(filename);
+
+            if (arguments != null)
+            {
+                processStartInfo.Arguments = arguments;
+            }
+
+            processStartInfo.StandardErrorEncoding = ProcessWrapper.OemEncoding;
```

Comment:
I'm extremely leery of this, especially since it always appears to be null. I *think* one of the encoding issues with gulp or npm disappeared when we switched to the new agent. Also, I never could repro the CopyFiles encoding issue on the new agent - even on the machine where we had the repro.